libs/wingc/src/type_check.rs

```diff
@@ -636,6 +636,28 @@ impl TypeRef {
 			_ => false,
 		}
 	}
+
+	pub fn is_serializable(&self) -> bool {
+		match **self {
+			Type::Resource(_) => false,
+			Type::Enum(_) => true,
+			Type::Number => true,
+			Type::String => true,
+			Type::Duration => true,
+			Type::Boolean => true,
+			Type::Json => true,
+			Type::Array(v) => v.is_serializable(),
+			Type::Map(v) => v.is_serializable(),
```

Comment:
I think that Map should not be valid. 

and maybe `is_serializable` is the wrong term to use. Enum is serializable but is not a json_value
maybe something like `is_legal_json_value`

Reply:
We need to leave map as valid. As for the name change I think thats fine.